packages/terra-notification-dialog/src/NotificationDialog.jsx

```diff
@@ -0,0 +1,184 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import AbstractModal from 'terra-abstract-modal';
+import Button from 'terra-button';
+import IconAlert from 'terra-icon/lib/icon/IconAlert';
+import IconError from 'terra-icon/lib/icon/IconError';
+import IconWarning from 'terra-icon/lib/icon/IconWarning';
+import IconInformation from 'terra-icon/lib/icon/IconInformation';
+import IconSuccess from 'terra-icon/lib/icon/IconSuccess';
+import classNames from 'classnames/bind';
+import 'terra-base/lib/baseStyles';
+import styles from './NotificationDialog.scss';
+
+const cx = classNames.bind(styles);
+
+const variants = {
+  ALERT: 'alert',
+  ERROR: 'error',
+  WARNING: 'warning',
+  INFO: 'info',
+  SUCCESS: 'success',
+  CUSTOM: 'custom',
+};
+
+const propTypes = {
+  /**
+   * Header of the notification-dialog.
+   */
+  header: PropTypes.string,
+  /**
+   * Title of the notification-dialog.
+   */
+  title: PropTypes.string,
+  /**
+   * Message of the notification-dialog.
+   */
+  message: PropTypes.string,
+  /**
+   * The Action of the primary button.
+   */
+  primaryAction: PropTypes.shape({
+    text: PropTypes.string,
+    onClick: PropTypes.func,
+  }),
+  /**
+   * The Action of the secondary button.
+   */
+  secondaryAction: PropTypes.shape({
+    text: PropTypes.string,
+    onClick: PropTypes.func,
+  }),
+  /**
+   * The variant of notification to be rendered.
+   * Use the Notification.Opts.Variants attribute of the Notification component for access to these variant strings.
```

Comment:
Let's update this callout the strings intended to be used with this. One of `alert`, `error`, `warning`, `info`, `success`, `custom` instead of using Notifications.Opts.Variants